### ISSUE TITLE
Inconsistent sidebar structure

### ISSUE THREAD

**Issue by Cerno-b** [godotengine/godot-docs#8535]
**Your Godot version: 4.2**

### Issue description:

Currently the sidebar TOC is inconsistent in some ways, and I would like to discuss possible solutions for this.

I can currently see 3 problems with the sidebar organisation:

1. Excessive nesting depth
    * leads to hidden and thus hard to access sections
2. Some sidebar headings have no own landing page
    * leads to confusing behavior where a user clicks a heading and lands on the page of its parent
3. Some pages show paragraphs in the sidebar, some don't
    * the decision about which paragraphs shows up does not seem to follow a discernible rule and can be confusing

#### Case 1

The current nesting is too deep. The sidebar allows a nesting depth of 5 (including ABOUT etc), but the actual nesting goes to 6 in some places. For example, the huge article about the [rendering architecture](https://docs.godotengine.org/en/latest/contributing/development/core_and_modules/internal_rendering_architecture.html) is completely inaccessible from the sidebar, and can only be read by knowing that there is a link in "Extending Godot by modifying its source code" or by cycling through the manual with the Next button.

![image](https://github.com/godotengine/godot-docs/assets/16470072/ba9023c1-60a1-41c7-86f4-18e714c530bd)

#### Case 2

Some sidebar headings have no own landing page, since they are just headings inside an `index.rst` file. Case in point: The [sidebar entry](https://docs.godotengine.org/en/latest/getting_started/first_2d_game/index.html#prerequisites) for `Your first 2D game > Prerequisites` lands on `Your first 2D game` and scrolls to the `Prerequisites` heading within that page since there is no separate `Prerequisites` landing page. I understand the motivation behind having the `Prerequisites` page in the sidebar, but there has to be a better way. In this case this is even more confusing since the paragraph the link points to is located at the end of the page so that the expected headline is not on top but in the middle of the screen:

![image](https://github.com/godotengine/godot-docs/assets/16470072/b0ac747e-af15-46a5-9fee-88b1fdd375ae)

#### Case 3

This is related to the previous case: Some sidebar entries point to specific landing pages, while others point to paragraphs within a landing page. This does not follow any discernible pattern and is structurally hard to follow, since a user might expect to find the heading of a previously read paragraph in the sidebar, because other paragraphs show up there as well.

### Proposal

I think it may be a good idea to enforce a few rules on the way the docs are structured, so they can be reviewed specifically as part of future structural changes.

#### Rule 1 

Every .rst page must be accessible through the sidebar.
This means that sidebar nesting levels >5 are forbidden

#### Rule 2

A sidebar heading must always refer to an rst file, never to a paragraph

#### Rationale

**Rule 1** should be fairly obvious and will resolve Case 1. The amount of work necessary is limited to the sections that are currently nested too deeply. I could attempt a PR with a reorg suggestion if this is desired. It would revolve around splitting off the topics about understanding and building the codebase from "Contributing" into their own main section.

**Rule 2** would resolve Case 2 and 3 but would cause some immediate structural problems that might only be resolvable through some heavy restructuring. 

For example, the `Your first 2D game > Prerequisites` section would have to be removed from the sidebar or moved to its own `rst` file, but it does not have enough material to warrant its own page. These kind of issues need to be reviewed and resolved.

This might require a lot of work, and it would probably only make sense to discuss the specifics of this if a first blueprint for such a restructuring was available. I would be interested in initializing a blueprint proposal as a basis for further discussion, but due to the work amount, I need some backing from the maintainers about whether this idea has the potential to follow up on.

**Comment by Cerno-b** [comment]
Or, if it's really important to have paragraph headings in the sidebar, at least make it consistent and use a maximum of 4 levels of depth for the rst files and add *all* first level headings to the sidebar. Still not ideal, since these paragraphs will appear at different sidebar levels and it wouldn't be immediately clear which are page links and which are paragraph links, but I wanted to point out the option.

**Comment by Cerno-b** [comment]
There is also the question of how to deal with intra-page TOCs which are not represented anywhere. I don't know what technical options we have, but I think it might be useful if we could provide access to the sections of a single .rst page instead of cramming them into the sidebar.

**Comment by YuriSizov** [comment]
Both of these rules mean we have to get rid of all organizational work that we have done by sectioning index files. This means we will end up with sections which are very long and disorganized and with index pages whose purpose is to only serve as a landing rather than be a part of a cohesive narrative.

I personally do not see any benefit in either of these two restrictions.

**Comment by Cerno-b** [comment]
@YuriSizov I don't know what has been done before since I have only recently arrived so I guess I was late to the party ;) I can evaluate the current state of things. Apologies if it seemed like I disregarded someone's work.

I understand that there is a lot of room for discussion for rule 2 and it may outright not work, but don't you see any merit in rule 1? Shouldn't we make sure that people can access every rst page from the sidebar? I think this is a serious problem. I don't think the proposal I have in mind will change much, content-wise for Case 1, it will just shift it around a bit to reduce nesting depth, while still maintaining the cohesive narrative. I am pretty confident I could work on a PR that reorganises parts of the "Contribution" section into its own section without changing the content too much. But I'd rather not do it if you are not on board, since you seem to have put a lot of though into fixing the problems that existed before the current state.

About rule 2, it is very possible that my proposal does not work out for the reasons you describe. But my question is: If I were to find a solution for the problems that come with a change like that, do you think it would have a chance to be accepted? As you said, apparently people have put in a significant amount of work and I don't want to step on anyone's toes. 

For me, who has recently started reading the manual from 0, it would have been a more pleasant experience if Case 2 and 3 would have been addressed in some way. That however ist just one person's opinion, and if I'm the only one, I'd rather not waste time drafting a change proposal nobody wants.

**Comment by Cerno-b** [comment]
Maybe a fairly simple example of what I would propose to change in order to resolve Case 2/3 and reduce nesting depth at the same time. It will be minimally invasive and improve navigation while not changing the narrative at all.

So please consider this: 

* Under `Your first 2D game`, we remove headings `Prerequisites` and `Contents` from the Sidebar.
* This moves all the content from the game tutorial one level up but changes nothing about the narrative. 
* Someone who clicks `Your first 2D game` will land at the top of the page and while reading it, will naturally get to prerequisites.
* The `Contents` section at the end of that page will still prepare the reader for what's to come, but it will mirror more naturally in the sidebar.

This is just one example. Could you let me know what you think of it? I believe it just makes up minor changes that open up space in the sidebar. If even this is not acceptable, I think I can stop here, but if this is workable, I can continue making minor change proposals like that.

**Comment by Cerno-b** [comment]
I just had a look at the official "Read the Docs" manual, which uses its own system, and it looks like the concept of using the sidebar to land in the middle of the page is a central idea in the system, so even if there was support for the idea, I don't think it would be reasonable to do. I'll just chalk it up as a "pet peeve" of mine and move on. The docs are usable as they are.

I still think we should find a solution to the hidden section in nesting level 6 as it's a shame that they are so hard to access when there is so much content, but I don' know what the best solution would be. Could be moving Code and Architecture parts to their own main section, could be getting rid of a grouping level somewhere in the middle or could be enabling a 6th sidebar level (I think I read in the Read the Docs that it's possible).

**Comment by Cerno-b** [comment]
I'm closing this issue because I think some of the ideas I raised are not reconcilable with the way Read the Docs works. I am opening a new issue for clarity that focuses on getting rid of deep nesting (https://github.com/godotengine/godot-docs/issues/8792)
